### PR TITLE
cloud: support to write/read azure blobs with customer provided key

### LINKED ISSUE #15024: support to write/read azure blobs with customer provided key
## Development Task
ref #14851 , we have supportted to write/read azure blobs with `encryption-scope`.

we need to also support to write/read azure blobs with customer provided key

### PR DESCRIPTION
**Pull request by Leavrth**
### What is changed and how it works?

Issue Number: Close #15024 

What's Changed:

```commit-message
support to write/read azure blobs with customer provided key
```

### Check List 

Tests 

- Unit test
- Manual test (add detailed scripts or steps below)
- No code

### Release note 

```release-note
None.
```

### CODE REVIEW

File: components/cloud/azure/src/azblob.rs [hunk 1]
```diff
@@ -339,6 +383,26 @@ impl AzureUploader {
...
+    fn adjust_put_builder(&self, builder: PutBlockBlobBuilder) -> PutBlockBlobBuilder {
+        // the encryption scope and the access tier can not be both in the HTTP headers
+        if let Some(scope) = &self.encryption_scope {
+            return builder.encryption_scope(scope.deref().clone());
+        }
+
+        // the encryption customer provided key and the access tier can not be both in
+        // the HTTP headers
+        if let Some(key) = &self.encryption_customer {
+            return builder.encryption_key::<(String, String)>(key.clone().into());
```
Comment (YuJuncen) [inline components/cloud/azure/src/azblob.rs:396]: So, it is impossible to provide both encryption scope and encryption key? Is this excepted?

**Review by YuJuncen — APPROVED** [review]
rest LGTM

File: components/cloud/azure/src/azblob.rs (hunk 1 above)
Comment (Leavrth) [reply components/cloud/azure/src/azblob.rs:396]: Yes, when both the encryption scope and the encryption key are in the headers, the azure server will return an error.

**Review by 3pointer — APPROVED** [review]
LGTM
But we need consider a question: what if user provide two ways of encryptions: encryption_scope and encryption_customer and one of them stop working occasional？

**Review by overvenus — APPROVED** [review]

**Pull request merged**